### ISSUE TITLE
Realm Object Server has issues with Capital letter email address as username

### ISSUE THREAD

**Issue by owenzhao** [realm/realm-object-server#159]
## Goals

Login to Realm Object Server from client app.  with 

```swift
let username = "WeightSync@foo.com"
let password = "weightsync"
```

## Expected Results

Login successfully

## Actual Results

Get an error `The provided credentials are invalid or a user does not exist`

## Steps to Reproduce

1. create a new user in Realm Object Server with username:"WeightSync@foo.com" and password:"weightsync"
2. add the username and password to client app
3. try to login

## Code Sample

```swift
SyncUser.logIn(with: .usernamePassword(username: username, password: password, register: false), server: UR...
```
## Version of Realm and Tooling

- Realm Object Server Version: 1.3.0
- Flavor: Developer
- Server OS & Version: macOS 10.12.4 (16E195)
- Client SDK Version: Realm master #aad71d2, Xcode:8.3 (8E162), Swift 3.1
- Client OS & Version: iOS 10.3(14E277)

## Logs

No need.

## Solution.

In Realm Object Server, the username with capital letter is treated as it is. However, on the client side, the username is converted to lowercased internally. That is why I can't login. Change either side will fix this issue.

**Comment by istx25** [comment]
Hi @owenzhao. Thanks for reaching out. I wanted to let you know that we've received your report and that someone will review what you've shared and follow-up soon with what we're going to do.

**Comment by bigfish24** [comment]
This should be fixed with ROS 2.0
